cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,197 @@
+		assert_eq!(reserved_bal, total_deposit);
+	});
+
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		let free_bal =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+		let reserved_bal = <PeopleRococo as PeopleRococoPallet>::Balances::reserved_balance(
+			PeopleRococoSender::get(),
+		);
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		// 3. Set identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			identity_parachain.clone()
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
+
+		// No events get triggered when calling set_sub_no_deposit
+
+		// No amount should be reserved as deposit amounts are set to 0.
+		let reserved_bal = <PeopleRococo as PeopleRococoPallet>::Balances::reserved_balance(
+			PeopleRococoSender::get(),
+		);
+		assert_eq!(reserved_bal, 0);
+		assert!(<PeopleRococo as PeopleRococoPallet>::Identity::identity(
+			&PeopleRococoSender::get()
+		)
+		.is_some());
+		let tuple_subs =
+			<PeopleRococo as PeopleRococoPallet>::Identity::subs_of(&PeopleRococoSender::get());
+		assert!(tuple_subs.1.len() > 0);
+	});
+
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		let reserved_balance = <RococoRelay as RococoRelayPallet>::Balances::reserved_balance(
+			RococoRelaySender::get(),
+		);
+		//before reap reserved balance should be equal to total deposit
+		assert_eq!(reserved_balance, total_deposit);
+		assert_ok!(<RococoRelay as RococoRelayPallet>::IdentityMigrator::reap_identity(
+			rococo_runtime::RuntimeOrigin::root(),
+			RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Unreserved { who, amount }) => {
+					who: *who == RococoRelaySender::get(),
+					amount: *amount == total_deposit,
+				},
+			]
+		);
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::identity(&RococoRelaySender::get())
+			.is_none());
+		let tuple_subs =
+			<RococoRelay as RococoRelayPallet>::Identity::subs_of(&RococoRelaySender::get());
+		assert_eq!(tuple_subs.1.len(), 0);
+
+		let reserved_balance = <RococoRelay as RococoRelayPallet>::Balances::reserved_balance(
+			RococoRelaySender::get(),
+		);
+		// after reap reserved balance should be 0
+		assert_eq!(reserved_balance, 0);
+	});
+
+	// 6. assert on Parachain
+	PeopleRococo::execute_with(|| {
+		let free_bal =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+
+		//Prior to reaping free balance was zero. If Free balance is > 0, then assets were
+		// teleported
+		assert!(free_bal > 0);
```

Comment:
We have these events for `Deposit` and `Withdraw`. In this case we should check that they match exactly, because it's supposed to be a feeless teleport.

Reply:
I've not checked the amounts due to the issue https://github.com/paritytech/polkadot-sdk/issues/2460 but the events are there. Once the bug is fixed I can add the additional checks.

Reply:
But why not check the amounts so that when the bug is fixed it automatically starts working?

Reply:
We can also remove this `assert!(free_bal > 0);` because it's redundant with `assert!(free_bal >= PEOPLE_ROCOCO_ED);` below.

Reply:
I'll check on the amounts 👌

Reply:
Added them for the `Reserved` events. When checking the values of `subs_deposit` and `id_deposit` they are the same as Reserved { amount }.

Also an `IdentityMigrator` event gets emitted with `subs_deposit` and `id_deposit`, these values are what we expect, but I can't assert IdentityMigrator events yet because of https://github.com/paritytech/polkadot-sdk/pull/2373#discussion_r1402661127